daemon/api_systems.go

```diff
@@ -104,19 +105,44 @@ func getAllSystems(c *Command, r *http.Request, user *auth.UserState) Response {
 }
 
 // wrapped for unit tests
-var deviceManagerSystemAndGadgetInfo = func(dm *devicestate.DeviceManager, systemLabel string) (*devicestate.System, *gadget.Info, error) {
-	return dm.SystemAndGadgetInfo(systemLabel)
+var deviceManagerSystemAndGadgetAndEncryptionInfo = func(dm *devicestate.DeviceManager, systemLabel string) (*devicestate.System, *gadget.Info, *devicestate.EncryptionSupportInfo, error) {
+	return dm.SystemAndGadgetAndEncryptionInfo(systemLabel)
+}
+
+func storageEncryption(encInfo *devicestate.EncryptionSupportInfo) *client.StorageEncryption {
+	storageEnc := &client.StorageEncryption{
+		StorageSafety: string(encInfo.StorageSafety),
+		Type:          string(encInfo.Type),
+	}
+	if encInfo.Disabled {
+		storageEnc.Support = "disabled"
+		return storageEnc
+	}
+	required := (encInfo.StorageSafety == asserts.StorageSafetyEncrypted)
+	switch {
+	case encInfo.Available:
+		storageEnc.Support = "available"
+	case !encInfo.Available && required:
```

Comment:
given how switches work the !Available part of these checks could be left out but maybe is more readable this way?

Reply:
I find it more readable this way but happy to change if other feels differently.

Reply:
I abstain :-)